pkg/kube/apis/extendedjob/v1alpha1/register.go

```diff
@@ -10,16 +13,116 @@ import (
 // This file looks almost the same for all controllers
 // Modify the addKnownTypes function, then run `make generate`
 
+const (
+	// ExtendedJobResourceKind is the kind name of ExtendedJob
+	ExtendedJobResourceKind = "ExtendedJob"
+	// ExtendedJobResourcePlural is the plural name of ExtendedJob
+	ExtendedJobResourcePlural = "extendedjobs"
+)
+
 var (
 	schemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
 
 	// AddToScheme is used for schema registrations in the controller package
 	// and also in the generated kube code
 	AddToScheme = schemeBuilder.AddToScheme
-)
 
-// SchemeGroupVersion is group version used to register these objects
-var SchemeGroupVersion = schema.GroupVersion{Group: apis.GroupName, Version: "v1alpha1"}
+	// ExtendedJobResourceShortNames is the short names of ExtendedJob
+	ExtendedJobResourceShortNames = []string{"ejob", "ejobs"}
+	// ExtendedJobValidation is the validation method for ExtendedJob
+	ExtendedJobValidation = extv1.CustomResourceValidation{
+		OpenAPIV3Schema: &extv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]extv1.JSONSchemaProps{
+				"spec": {
+					Type: "object",
+					Properties: map[string]extv1.JSONSchemaProps{
+						"output": {
+							Type: "object",
+							Properties: map[string]extv1.JSONSchemaProps{
+								"namePrefix": {
+									Type: "string",
+								},
+								"outputType": {
+									Type: "string",
+								},
+								"secretLabels": {
+									Type: "object",
+								},
+								"writeOnFailure": {
+									Type: "boolean",
+								},
+							},
+							Required: []string{
+								"namePrefix",
+							},
+						},
+						"trigger": {
+							Type: "object",
+							Properties: map[string]extv1.JSONSchemaProps{
+								"strategy": {
+									Type: "string",
+									Enum: []extv1.JSON{
+										{
+											Raw: []byte(`"manual"`),
+										},
+										{
+											Raw: []byte(`"once"`),
+										},
+										{
+											Raw: []byte(`"now"`),
+										},
+										{
+											Raw: []byte(`"podstate"`),
```

Comment:
We no longer support podstate, should have been deleted when trigger ejobs were removed.

Reply:
done